packages/core/src/document/documentStore.ts

```diff
@@ -183,8 +183,21 @@ const _getDocumentState = bindActionByDataset(
   documentStore,
   createStateSourceAction({
     selector: ({state: {error, documentStates}}, options: DocumentOptions<string | undefined>) => {
-      const {documentId, path} = options
+      const {documentId, path, liveEdit} = options
       if (error) throw error
+
+      if (liveEdit) {
+        // For liveEdit documents, only look at the single document
+        const document = documentStates[documentId]?.local
+        if (document === undefined) return undefined
+        if (!path) return document
```

Comment:
Extremely nit picky but since of both of these checks return the value of `document` they could be combined. Something like:

```suggestion
        if (document === undefined || !path) return document
```